index.bs

```diff
@@ -10983,42 +10978,46 @@ Note: Since a legacy callback interface object is a [=function object=]
+    with [=identifier=] |id| and in [=Realm=] |realm|
+    is <dfn lt="create a legacy callback interface object">created</dfn> as follows:
 
     1.  Let |steps| be the following steps:
         1.  [=ECMAScript/Throw=] a {{ECMAScript/TypeError}}.
-    1.  Let |F| be [=!=] <a abstract-op>CreateBuiltinFunction</a>(|realm|, |steps|, the {{%FunctionPrototype%}} of |realm|).
+    1.  Let |F| be [=!=] <a abstract-op>CreateBuiltinFunction</a>(|realm|, |steps|, |realm|.\[[Intrinsics]].[[{{%FunctionPrototype%}}]]).
     1.  Perform [=!=] <a abstract-op>SetFunctionName</a>(|F|, |id|).
     1.  Perform [=!=] <a abstract-op>DefinePropertyOrThrow</a>(|F|, "<code>length</code>",
         PropertyDescriptor{\[[Value]]: 0, \[[Writable]]: <emu-val>false</emu-val>, \[[Enumerable]]: <emu-val>false</emu-val>, \[[Configurable]]: <emu-val>true</emu-val>}).
+    1.  [=Define the constants=] of |interface| on |F| given |realm|.
 </div>
 
 
 <h4 id="named-properties-object">Named properties object</h4>
 
 For every [=interface=] declared with the [{{Global}}] [=extended attribute=]
 that [=support named properties|supports named properties=],
-there must exist an object known as the
-<dfn id="dfn-named-properties-object" export>named properties object</dfn>
+there must exist an object known as the <dfn id="dfn-named-properties-object" export>named properties object</dfn>
 for that interface on which named properties are exposed.
 
-<div algorithm="value of [[Prototype]] property of named properties objects">
-
-    The [=named properties object=]
-    for a given interface |A| must have a
-    \[[Prototype]] [=internal slot=] whose value is returned from
-    the following steps:
-
-    1.  If |A| is declared to inherit from another interface, then return the
-        [=interface prototype object=]
-        for the inherited interface.
-    1.  Otherwise, if |A| is declared with the [{{LegacyArrayClass}}]
-        extended attribute, then return {{%ArrayPrototype%}}.
-    1.  Otherwise, return {{%ObjectPrototype%}}.
+<div algorithm>
+    The [=named properties object=] for a given [=interface=] |interface| and [=Realm=] |realm|,
+    is <dfn lt="create a named properties object">created</dfn> as follows:
+
+    1.  Let |proto| be null.
+    1.  If |interface| is declared to inherit from another interface,
+        then set |proto| to the [=interface prototype object=] in |realm| for the inherited interface.
+    1.  Otherwise, if |interface| is declared with the [{{LegacyArrayClass}}] [=extended attribute=],
+        then set |proto| to |realm|.\[[Intrinsics]].[[{{%ArrayPrototype%}}]].
+    1.  Otherwise, set |proto| to |realm|.\[[Intrinsics]].[[{{%ObjectPrototype%}}]].
+    1.  Let |obj| be a newly created object.
+    1.  Set |obj|'s internal methods to the definitions specified in
+        [=ECMA-262 §9.1|ECMA-262 §9.1 Ordinary object internal methods and internal slots=], unless they are specified below.
```

Comment:
Maybe s/below/the rest of this section, or even the rest of `[[#named-properties-object]]`?

Reply:
Fixed in https://github.com/heycam/webidl/pull/494/commits/727287a754551e961546524d7f5631d9191f26ef.